### DISCUSSION TITLE
How to call nickel interpreter from python?

### DISCUSSION THREAD

**Discussion by simkimsia** [tweag/nickel#601] [Q&A]
I saw in the read me it says can call interpreter from many programming languages

So I was wondering if I can call the nickel interpreter to generate a yaml from inside a python script?

**Comment by silverraven691** [comment]
At the moment, the easiest would be to run the Nickel CLI in a subprocess.
If you feel adventurous, you could use the WASM module that's built for the website, probably by inspecting how the website works https://github.com/tweag/nickel-lang.org

**Reply by yannham** [reply]
I second @silverraven691. That being said, we chose Rust especially because it has no runtime and it should be possible to interface it with other languages through a FFI. If running Nickel as a separate process is not enough for any reason, feel free to open an issue as a feature request for Python bindings.
